### PR TITLE
Code coverage: Loader service - 100%

### PR DESCRIPTION
**Pull request by Dimple16**
Screenshot 2023-02-20 at 11 19 07 PM

### CODE REVIEW

File: src/app/core/services/loader.service.spec.ts [hunk 1]
```diff
@@ -1,16 +1,63 @@
...
+  describe('showLoader()', () => {
+    it('showLoader(): should show loader with args', fakeAsync(() => {
+      const message = 'Please wait...';
+      const duration = 1000;
+      const loadingElementSpy = jasmine.createSpyObj('HTMLIonLoadingElement', ['present']);
+      loadingController.create.and.returnValue(Promise.resolve(loadingElementSpy));
+
+      service.showLoader(message, duration);
+      tick();
+      expect(loadingController.create).toHaveBeenCalledWith({
```
Comment (rvab) [inline src/app/core/services/loader.service.spec.ts:36]: `toHaveBeenOnceCalledWith`?

File: src/app/core/services/loader.service.spec.ts [hunk 2]
```diff
@@ -1,16 +1,63 @@
...
+        message,
+        duration,
+      });
+      expect(loadingElementSpy.present).toHaveBeenCalled();
```
Comment (rvab) [inline src/app/core/services/loader.service.spec.ts:40]: `toHaveBeenCalledTimes`?

File: src/app/core/services/loader.service.spec.ts [hunk 3]
```diff
@@ -1,16 +1,63 @@
...
+    }));
+
+    it('showLoader(): should show loader without args', fakeAsync(() => {
+      const loadingElementSpy = jasmine.createSpyObj('HTMLIonLoadingElement', ['present']);
+      loadingController.create.and.returnValue(Promise.resolve(loadingElementSpy));
+
+      service.showLoader();
+      tick();
+      expect(loadingController.create).toHaveBeenCalled();
+      expect(loadingElementSpy.present).toHaveBeenCalled();
```
Comment (rvab) [inline src/app/core/services/loader.service.spec.ts:50]: `toHaveBeenCalledTimes(1)` for both?

File: src/app/core/services/loader.service.spec.ts [hunk 4]
```diff
@@ -1,16 +1,63 @@
...
+    }));
+  });
+
+  it('hideLoader(): should hide loader', fakeAsync(() => {
```
Comment (rvab) [inline src/app/core/services/loader.service.spec.ts:54]: there's a catch block, no? how is it covered?

**Review by rvab — CHANGES_REQUESTED** [review]

File: src/app/core/services/loader.service.spec.ts (hunk 4 above)
Comment (Dimple16) [reply src/app/core/services/loader.service.spec.ts:54]: added a test case to cover it

**Review by rvab — APPROVED** [review]

**Pull request merged**